src/goto-instrument/contracts/contracts.cpp

```diff
@@ -171,33 +223,54 @@ void code_contractst::check_apply_loop_contracts(
       new_tmp_symbol(
         clause.type(), loop_head->source_location(), mode, symbol_table)
         .symbol_expr();
-    havoc_code.add(goto_programt::make_decl(
+    generated_code.add(goto_programt::make_decl(
       old_decreases_var, loop_head->source_location()));
     old_decreases_vars.push_back(old_decreases_var);
 
     const auto new_decreases_var =
       new_tmp_symbol(
         clause.type(), loop_head->source_location(), mode, symbol_table)
         .symbol_expr();
-    havoc_code.add(goto_programt::make_decl(
+    generated_code.add(goto_programt::make_decl(
       new_decreases_var, loop_head->source_location()));
     new_decreases_vars.push_back(new_decreases_var);
   }
 
-  // non-deterministically skip the loop if it is a do-while loop
+  // non-deterministically skip the loop if it is a do-while loop.
+  // TODO: Fix loop contract issues for do/while loops.
```

Comment:
Let's create an Github issue to make sure we track this request and add the link here.
Shall we just stop applying invariants to do-while loops and report "not supported"?

Reply:
That sounds like a good plan. I'll make this change. Thanks.